staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -123,6 +124,9 @@ func (s *ServerRunOptions) DefaultAdvertiseAddress(secure *SecureServingOptions)
 func (s *ServerRunOptions) Validate() []error {
 	errors := []error{}
 
+	if s.AdvertisePort < 0 || s.AdvertisePort > 65535 {
```

Comment:
we should avoid zero, it will allocate one random port

Reply:
This is specifically used as the advertise address and _not_ the bind address (i.e. it's what is stored/persisted into the Endpoint/EndpointSlice). So zero doesn't make sense in this context besides being the empty/zero value.

Reply:
If we were to turn this into a pointer and disambiguate `0` from `nil`, we'd then also presumably mark `0` as an invalid input (as it doesn't make sense to set the port number to zero in the Endpoint[Slice] object, unless it _is_ possible for a process to bind on port 0.. which as far as I am aware it is not)